ggml/src/CMakeLists.txt

```diff
@@ -486,9 +486,11 @@ if (GGML_SYCL)
     add_compile_options(-I./) #include DPCT
 
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-narrowing")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
     if (GGML_SYCL_TARGET STREQUAL "NVIDIA")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl-targets=nvptx64-nvidia-cuda")
+        add_compile_definitions(GGML_SYCL_WARP_SIZE=32)
+    else()
+        add_compile_definitions(GGML_SYCL_WARP_SIZE=16)
```

Comment:
@qnixsynapse Could you change this size to 32 and test your model?

Reply:
Works perfectly on warp size of 32!!!! 
![image](https://github.com/ggerganov/llama.cpp/assets/17230826/92db5f51-c486-4fb4-a4cf-c007866c7173)